### PR TITLE
Timeout

### PR DESCRIPTION
**Pull request by iandewancker**
This is for issue 337 :  https://github.com/joblib/joblib/issues/337

### CODE REVIEW

File: joblib/test/test_parallel.py [hunk 1]
```diff
@@ -173,6 +184,23 @@ def g(x):
                             )
 
 
+def test_parallel_timeout_success():
+    # Check that timeout isn't thrown when function is fast enough
+    test_lst = [x ** 2 for x in range(10)]
+    nose.tools.assert_equal(
+               test_lst,
+               Parallel(n_jobs=2, timeout=3)(delayed(fast_f)(x) for x in range(10))
```
Comment (ogrisel) [inline joblib/test/test_parallel.py]: No need to introduce helper functions for this, you can just do:

```python
nose.tools.assert_equal(
    [x ** 2 for x in range(10)],
    Parallel(n_jobs=2, timeout=10)(delayed(sleep)(0.001) for x in range(10)))
```

File: joblib/test/test_parallel.py [hunk 2]
```diff
@@ -173,6 +184,23 @@ def g(x):
...
+              )
+
+
+def test_parallel_timeout_fail():
+    # Check that timeout properly fails when function is too slow
+    nose.tools.assert_raises(TimeoutError,
+                             Parallel(n_jobs=2, timeout=5),
+                             (delayed(slow_f)(x) for x in range(10))
+                            )
```
Comment (ogrisel) [inline joblib/test/test_parallel.py]: Please make the tests run as fast as possible (while still testing the behavior you are interested in:

```python
nose.tools.assert_raises(TimeoutError,
    Parallel(n_jobs=2, timeout=0.01),
    (delayed(sleep)(10) for x in range(10))
 )
```

**Comment by ogrisel** [comment]
The continuous integration servers cannot run on this PR because this branch cannot be merged to master (there are conflicts).

You need to rebase your changes on top of the current master (and resolve conflicts and then force push your changes to update this PR). Please do not merge the upstream master into your branch but instead rebase your branch on top of an up-to-date master. We appreciate keeping a linear commit history in the master branch to keep it easy to audit.

**Comment by lesteve** [comment]
Closing in favour of #340.

**Pull request closed without merging**